packages/storybook/stories/utils.ts

```diff
@@ -1,25 +1,135 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-export const renderVideoStream = (scalingMode?: string): HTMLElement | null => {
+import { useEffect, useRef, useState } from 'react';
+
+const renderVideoStream = async (scalingMode?: string): Promise<HTMLElement | null> => {
   if (navigator.mediaDevices?.getUserMedia) {
     const video = document.createElement('video');
     video.autoplay = true;
     video.style.height = '100%';
     video.style.width = '100%';
     video.style.objectFit = scalingMode ?? 'cover';
 
-    navigator.mediaDevices
-      .getUserMedia({ video: true })
-      .then(function (stream) {
-        video.srcObject = stream;
-      })
-      .catch(function (error) {
-        console.log('Something went wrong!', error);
-      });
+    try {
+      video.srcObject = await navigator.mediaDevices.getUserMedia({ video: true });
+    } catch (e) {
+      console.log('Unable to get stream for video', e);
+    }
 
     return video;
   }
 
   return null;
 };
+
+const stopRenderVideoStream = (video: HTMLElement | null): void => {
+  if (!video) {
+    return;
+  }
+
+  const mediaProvider = (video as HTMLVideoElement).srcObject;
+  if (!mediaProvider) {
+    return;
+  }
+
+  const stream = mediaProvider as MediaStream;
+  if (!stream.getVideoTracks) {
+    return;
+  }
+
+  for (const track of stream.getVideoTracks()) {
+    track.stop();
+  }
+};
+
+/**
+ * Manages a single video stream. If enabled is true will create and return a video stream. When component is unmounted
+ * will automatically dispose of the video stream.
+ *
+ * @param enabled - If creating a video stream
+ * @returns
+ */
+export const useVideoStream = (enabled: boolean): HTMLElement | null => {
+  const mounted = useRef(false);
+  useEffect(() => {
+    mounted.current = true;
+    return () => {
+      mounted.current = false;
+    };
+  });
+
+  const [videoStreamElement, setVideoStreamElement] = useState<HTMLElement | null>(null);
+  useEffect(() => {
+    const loadVideo = async (): Promise<void> => {
+      const videoStreamElement = await renderVideoStream();
```

Comment:
prefer using a different variable name to avoid shadowing `videoStreamElement` from line 62.
Especially because `setVideoStreamElement` is used below, it's confusing that only one of the variables defined on line 62 is shadowed.

Reply:
Sounds good, I've removed this function actually as per your comment below about re-using the hook code.